### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by GedeonTS**
## TDD project

In this PR me @GedeonTS and my partner @kwambiee  we :
- created test with respec_mocks for the methods below:
- [ ] factorial
- [ ] reverse
- [ ] fizzBuzz
- then created :
- [ ] a class called `Solver`.
- [ ] a method called `factorial` that takes one argument, an integer `N`, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to `N` and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- [ ] a method called `reverse` that takes one argument, a string `word`, and returns `word` reversed (e.g. if `word` is `"hello"` it returns `"olleh"`).
- [ ] a method called `fizzbuzz` that takes one argument, an integer `N`, and returns a string. The returned string is constructed following these rules:
- [ ] When `N` is divisible by 3, return `"fizz"`.
- [ ] When `N` is divisible by 5, return `"buzz"`.
- [ ] When `N` is divisible by 3 and 5, return `"fizzbuzz"`.
- [ ] Any other case, return `N` as a string (e.g. say `N` is `7` then return `"7"`).
- [ ] Remember to start with tests for all of the methods described above. Your commit history should prove that you have used TDD.

Your review is welcome! :smile:

### CODE REVIEW

File: solver.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+        if number < 0
+            raise ArgumentError
+        end
+        if number == 0
+            1
+        else
+            number * factorial(number - 1)
+        end
+    end
+end
```
Comment (Kingstalux) [inline solver.rb]: - [ ] Good job so far :+1: but currently all your tests are failing. Kindly fix that and make sure your tests are passing before submitting for a review :smile: 
- Your tests:
Screenshot from 2022-07-28 13-16-33

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @GedeonTS and @kwambiee, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights :clap: 
:ballot_box_with_check: Good commit history :+1: 
:ballot_box_with_check: Linters are passing :ok_hand: 
:ballot_box_with_check: Good job writing tests before writing the functions :100: 
:ballot_box_with_check: Professional README file :clap: 



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lordkaito — APPROVED** [review]
Hi @GedeonTS, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by GedeonTS** [comment]
Thanks @Lordkaito !

**Pull request merged**